WordPress/Classes/Networking/MediaHost+ReaderPostContentProvider.swift

```diff
@@ -0,0 +1,25 @@
+import Foundation
+
+/// Extends `MediaRequestAuthenticator.MediaHost` so that we can easily
+/// initialize it from a given `Blog`.
+///
+extension MediaHost {
```

Comment:
I'm thinking about the three `MediaHost` extensions and wondering if it would be better to have a single `MediaHost+Initializers` extension for all rather than three separate ones that do sort of the same thing? The separate files seem a tad too granular? 
Up to you, just something to consider.

Reply:
They were in the same file at first, but I ended up separating them because I am planning to add more tests (in an upcoming PR) for each  of these initializers.  I usually try to put the tests inside files matching the tested files...  so for example I plan to have:

`MediaHostAbstractPostTests.swift`
`MediaHostBlogTests.swift`
`MediaHostReaderPostContentProvider.swift`

With around 8 tests each.

Reply:
Ah, I should have figured there was something more behind the decision :)